src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -769,6 +769,254 @@ public static void Compare_LongString()
             Assert.True(result < 0);
         }
 
+        [Fact]
+        public static void ZeroLengthCompareTo_StringComparison()
+        {
+            string value = "456";
+            string s = value.Substring(2, 0);
+
+            Assert.True(0 < string.Compare(value, s, StringComparison.Ordinal));
+
+            Assert.True(0 < string.Compare(value, s, StringComparison.CurrentCulture));
+            Assert.True(0 < string.Compare(value, s, StringComparison.CurrentCultureIgnoreCase));
+            Assert.True(0 < string.Compare(value, s, StringComparison.InvariantCulture));
+            Assert.True(0 < string.Compare(value, s, StringComparison.InvariantCultureIgnoreCase));
+            Assert.True(0 < string.Compare(value, s, StringComparison.OrdinalIgnoreCase));
+
+            string emptyValue = value.Substring(1, 0);
+            Assert.Equal(0, string.Compare(emptyValue, s, StringComparison.Ordinal));
+
+            Assert.Equal(0, string.Compare(emptyValue, s, StringComparison.CurrentCulture));
+            Assert.Equal(0, string.Compare(emptyValue, s, StringComparison.CurrentCultureIgnoreCase));
+            Assert.Equal(0, string.Compare(emptyValue, s, StringComparison.InvariantCulture));
+            Assert.Equal(0, string.Compare(emptyValue, s, StringComparison.InvariantCultureIgnoreCase));
+            Assert.Equal(0, string.Compare(emptyValue, s, StringComparison.OrdinalIgnoreCase));
+
+            char[] a = value.ToArray();
+
+            var span = new ReadOnlySpan<char>(a);
```

Comment:
How about using `var span = value.AsSpan` instead of `new ReadOnySpan<char>(value.ToArray())`? Same with other spans of chars.